FAIL: TestPreallocate: preallocate_test.go:29: unexpected error: interrupted system call

Output of `restic version`
--------------------------
`restic 0.17.1 compiled with go1.23.1 on linux/arm64`


Problem description / Steps to reproduce
----------------------------------------
One test failed accidentally on aarch64; the next run did not show the problem.

```
[00:00:51] + RESTIC_TEST_FUSE=0
[00:00:51] + go test ./...
[00:00:56] ?   	github.com/restic/restic	[no test files]
[00:01:09] ?   	github.com/restic/restic/helpers/build-release-binaries	[no test files]
[00:01:09] ?   	github.com/restic/restic/helpers/prepare-release	[no test files]
[00:01:10] ?   	github.com/restic/restic/internal/backend/logger	[no test files]
[00:01:10] ?   	github.com/restic/restic/internal/backend/mock	[no test files]
[00:01:11] ?   	github.com/restic/restic/internal/backend/test	[no test files]
[00:01:13] ?   	github.com/restic/restic/internal/test	[no test files]
[00:01:13] ?   	github.com/restic/restic/internal/ui/signals	[no test files]
[00:01:26] ok  	github.com/restic/restic/cmd/restic	13.984s
[00:01:26] ok  	github.com/restic/restic/internal/archiver	2.333s
[00:01:26] ok  	github.com/restic/restic/internal/backend	0.521s
[00:01:26] ok  	github.com/restic/restic/internal/backend/azure	0.007s
[00:01:26] ok  	github.com/restic/restic/internal/backend/b2	0.007s
[00:01:26] ok  	github.com/restic/restic/internal/backend/cache	0.609s
[00:01:26] ok  	github.com/restic/restic/internal/backend/dryrun	0.005s
[00:01:26] ok  	github.com/restic/restic/internal/backend/gs	0.012s
[00:01:26] ok  	github.com/restic/restic/internal/backend/layout	0.021s
[00:01:26] ok  	github.com/restic/restic/internal/backend/limiter	0.046s
[00:01:26] ok  	github.com/restic/restic/internal/backend/local	2.006s
[00:01:26] ok  	github.com/restic/restic/internal/backend/location	0.006s
[00:01:26] ok  	github.com/restic/restic/internal/backend/mem	2.119s
[00:01:26] ok  	github.com/restic/restic/internal/backend/rclone	0.008s
[00:01:26] ok  	github.com/restic/restic/internal/backend/rest	0.011s
[00:01:26] ok  	github.com/restic/restic/internal/backend/retry	0.056s
[00:01:26] ok  	github.com/restic/restic/internal/backend/s3	0.008s
[00:01:26] ok  	github.com/restic/restic/internal/backend/sema	0.011s
[00:01:26] ok  	github.com/restic/restic/internal/backend/sftp	0.007s
[00:01:26] ok  	github.com/restic/restic/internal/backend/swift	0.007s
[00:01:26] ok  	github.com/restic/restic/internal/backend/util	0.007s
[00:01:26] ok  	github.com/restic/restic/internal/bloblru	0.006s
[00:01:26] ok  	github.com/restic/restic/internal/checker	2.385s
[00:01:26] ok  	github.com/restic/restic/internal/crypto	0.399s
[00:01:26] ok  	github.com/restic/restic/internal/debug	0.007s
[00:01:26] ok  	github.com/restic/restic/internal/dump	0.155s
[00:01:26] ok  	github.com/restic/restic/internal/errors	0.004s
[00:01:26] ok  	github.com/restic/restic/internal/feature	0.004s
[00:01:26] ok  	github.com/restic/restic/internal/filter	0.465s
[00:01:26] --- FAIL: TestPreallocate (0.00s)
[00:01:26]     --- FAIL: TestPreallocate/1 (0.00s)
[00:01:26]         preallocate_test.go:29: [31munexpected error: interrupted system call[39m
[00:01:26]             
[00:01:26] FAIL
[00:01:26] FAIL	github.com/restic/restic/internal/fs	0.838s
[00:01:26] ok  	github.com/restic/restic/internal/fuse	0.083s
[00:01:26] ok  	github.com/restic/restic/internal/migrations	0.010s
[00:01:26] ok  	github.com/restic/restic/internal/options	0.005s
[00:01:26] ok  	github.com/restic/restic/internal/repository	4.809s
[00:01:26] ok  	github.com/restic/restic/internal/repository/hashing	0.020s
[00:01:26] ok  	github.com/restic/restic/internal/repository/index	0.685s
[00:01:26] ok  	github.com/restic/restic/internal/repository/pack	0.011s
[00:01:26] ok  	github.com/restic/restic/internal/restic	0.576s
[00:01:26] ok  	github.com/restic/restic/internal/restorer	0.762s
[00:01:26] ok  	github.com/restic/restic/internal/selfupdate	0.008s
[00:01:26] ok  	github.com/restic/restic/internal/textfile	0.004s
[00:01:26] ok  	github.com/restic/restic/internal/ui	0.004s
[00:01:26] ok  	github.com/restic/restic/internal/ui/backup	0.019s
[00:01:26] ok  	github.com/restic/restic/internal/ui/progress	0.214s
[00:01:26] ok  	github.com/restic/restic/internal/ui/restore	0.007s
[00:01:26] ok  	github.com/restic/restic/internal/ui/table	0.005s
[00:01:26] ok  	github.com/restic/restic/internal/ui/termstatus	0.008s
[00:01:26] ok  	github.com/restic/restic/internal/walker	0.010s
[00:01:26] FAIL
```

Do you have any idea what may have caused this?
-----------------------------------------------
`EINTR` not handled properly?

--- a/internal/fs/preallocate_linux.go
+++ b/internal/fs/preallocate_linux.go
@@ -2,6 +2,7 @@ package fs
 
 import (
 	"os"
+	"syscall"
 
 	"golang.org/x/sys/unix"
 )
@@ -12,5 +13,17 @@ func PreallocateFile(wr *os.File, size int64) error {
 	}
 	// int fallocate(int fd, int mode, off_t offset, off_t len)
 	// use mode = 0 to also change the file size
-	return unix.Fallocate(int(wr.Fd()), 0, 0, size)
+	return ignoringEINTR(func() error { return unix.Fallocate(int(wr.Fd()), 0, 0, size) })
+}
+
+// ignoringEINTR makes a function call and repeats it if it returns
+// an EINTR error.
+// copied from /usr/lib/go/src/internal/poll/fd_posix.go of go 1.23.1
+func ignoringEINTR(fn func() error) error {
+	for {
+		err := fn()
+		if err != syscall.EINTR {
+			return err
+		}
+	}
 }

